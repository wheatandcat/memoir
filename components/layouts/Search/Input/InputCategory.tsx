import { memo, type FC } from 'react';
import { StyleSheet } from 'react-native';
import View from '@/components/elements/View';
import setting from '@/components/elements/Category/setting';
import CategoryButton from '@/components/layouts/CategoryButton';

export type Props = {
  categoryID: number;
  onPress: (categoryID: number) => void;
};

const InputCategory: FC<Props> = (props) => {
  const category1 = setting().main[0];
  const category2 = setting().main[1];
  const category3 = setting().main[2];

  return (
    <>
      <View style={styles.category}>
        {category1.map((v) => (
          <View key={v} mx={2}>
            <CategoryButton
              categoryID={v}
              selected={false}
              opacity={!(props.categoryID === v)}
              onPress={() => props.onPress(v)}
              displayName={false}
            />
          </View>
        ))}
      </View>
      <View style={styles.category}>
        {category2.map((v) => (
          <View key={v} mx={2}>
            <CategoryButton
              categoryID={v}
              selected={false}
              opacity={!(props.categoryID === v)}
              onPress={() => props.onPress(v)}
              displayName={false}
            />
          </View>
        ))}
      </View>
      <View style={styles.category}>
        {category3.map((v) => (
          <View key={v} mx={2}>
            <CategoryButton
              categoryID={v}
              selected={false}
              opacity={!(props.categoryID === v)}
              onPress={() => props.onPress(v)}
              displayName={false}
            />
          </View>
        ))}
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  category: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
});

export default memo(InputCategory);
