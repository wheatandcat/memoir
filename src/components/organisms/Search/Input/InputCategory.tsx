import React, { memo } from 'react';
import { StyleSheet } from 'react-native';
import View from 'components/atoms/View';
import setting from 'components/atoms/Category/setting';
import CategoryButton from 'components/molecules/CategoryButton';

export type Props = {};

const InputCategory: React.FC<Props> = () => {
  const category1 = setting().main[0];
  const category2 = setting().main[1];
  const category3 = setting().main[2];

  return (
    <>
      <View style={styles.category}>
        {category1.map((v) => (
          <View key={v} mx={2}>
            <CategoryButton
              categoryID={v}
              selected={false}
              opacity
              onPress={() => null}
              displayName={false}
            />
          </View>
        ))}
      </View>
      <View style={styles.category}>
        {category2.map((v) => (
          <View key={v} mx={2}>
            <CategoryButton
              categoryID={v}
              selected={false}
              opacity
              onPress={() => null}
              displayName={false}
            />
          </View>
        ))}
      </View>
      <View style={styles.category}>
        {category3.map((v) => (
          <View key={v} mx={2}>
            <CategoryButton
              categoryID={v}
              selected={false}
              opacity
              onPress={() => null}
              displayName={false}
            />
          </View>
        ))}
      </View>
    </>
  );
};

const styles = StyleSheet.create({
  category: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
});

export default memo(InputCategory);
