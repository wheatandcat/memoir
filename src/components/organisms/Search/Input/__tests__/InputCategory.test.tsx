import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import InputCategory, { Props } from '../InputCategory';

const propsData = (): Props => ({});

describe('components/organisms/Search/Input/InputCategory.tsx', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<InputCategory {...propsData()} />);
  });

  it('正常にrenderすること', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
