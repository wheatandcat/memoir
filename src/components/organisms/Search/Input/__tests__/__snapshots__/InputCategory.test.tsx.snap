// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/Search/Input/InputCategory.tsx 正常にrenderすること 1`] = `
<Fragment>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      key="1"
      mx={2}
    >
      <Memo(CategoryButton)
        categoryID={1}
        displayName={false}
        onPress={[Function]}
        opacity={true}
        selected={false}
      />
    </View>
    <View
      key="10"
      mx={2}
    >
      <Memo(CategoryButton)
        categoryID={10}
        displayName={false}
        onPress={[Function]}
        opacity={true}
        selected={false}
      />
    </View>
    <View
      key="4"
      mx={2}
    >
      <Memo(CategoryButton)
        categoryID={4}
        displayName={false}
        onPress={[Function]}
        opacity={true}
        selected={false}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      key="3"
      mx={2}
    >
      <Memo(CategoryButton)
        categoryID={3}
        displayName={false}
        onPress={[Function]}
        opacity={true}
        selected={false}
      />
    </View>
    <View
      key="2"
      mx={2}
    >
      <Memo(CategoryButton)
        categoryID={2}
        displayName={false}
        onPress={[Function]}
        opacity={true}
        selected={false}
      />
    </View>
    <View
      key="5"
      mx={2}
    >
      <Memo(CategoryButton)
        categoryID={5}
        displayName={false}
        onPress={[Function]}
        opacity={true}
        selected={false}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      key="13"
      mx={2}
    >
      <Memo(CategoryButton)
        categoryID={13}
        displayName={false}
        onPress={[Function]}
        opacity={true}
        selected={false}
      />
    </View>
    <View
      key="6"
      mx={2}
    >
      <Memo(CategoryButton)
        categoryID={6}
        displayName={false}
        onPress={[Function]}
        opacity={true}
        selected={false}
      />
    </View>
    <View
      key="12"
      mx={2}
    >
      <Memo(CategoryButton)
        categoryID={12}
        displayName={false}
        onPress={[Function]}
        opacity={true}
        selected={false}
      />
    </View>
  </View>
</Fragment>
`;
